politeiad/backendv2/tstorebe/store/localdb/localdb.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package localdb
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/store"
+	"github.com/decred/politeia/util"
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+var (
+	_ store.BlobKV = (*localdb)(nil)
+)
+
+// localdb implements the store BlobKV interface using leveldb.
+type localdb struct {
+	sync.Mutex
+	shutdown bool
+	db       *leveldb.DB
+
+	// Encryption key and mutex. The key is zero'd out on application
+	// exit so the read lock must be held during concurrent access to
+	// prevent the golang race detector from complaining.
+	key    *[32]byte
+	keyMtx sync.RWMutex
```

Comment:
Do we really want two mutexes?

Reply:
The named mutex has been removed and replaced with a unnamed mutex.

https://github.com/decred/politeia/pull/1180/files#diff-9cbdfdfae5a29aa39168b4484642d0acb8fdd880dcccdb358538928ac60cdb70R47